src/vcf/record.cpp

```diff
@@ -679,6 +673,60 @@ namespace ebi
                 message << "INFO " << RUB << " for record at " << line << " must have " + RUC;
                 throw new InfoBodyError{line, message.str()};
             }
+        } else if (field_key == MEINFO) {
+            if (source->version < Version::v44) {   //not applicable for anything < v4.4
+                return;
+            }
+            if (values.size() != 4 * alternate_alleles.size()) {            //MEINFO must be 4 * ALT allele count
+                std::stringstream message;
+                message << "INFO " << MEINFO << " for record at " << line << " must have " << 4 * alternate_alleles.size() << " value(s)";
+                throw new InfoBodyError{line, message.str(), "Found " + std::to_string(values.size()) + " value(s)"};
+            }
```

Comment:
I feel like this should be refactored as a function to be used when cardinality is X time ALT.